src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
+        int keyHash = (key == null) ? 0 : key.hashCode();
+        Node<K, V> newNode = new Node<>(keyHash, key, value, null);
+        int index = hash(key);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (Objects.equals(key, current.key)) {
+                    current.value = value;
+                    return;
+                } else if (current.next == null) {
+                    current.next = newNode;
+                    size++;
+                    break;
+                }
+                current = current.next;
+            }
 
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> current = table[index];
+        while (current != null) {
+            if (Objects.equals(key, current.key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    public void resize() {
```

Comment:
Consider renaming the `resize` method to `resize` to adhere to Java naming conventions. This will improve code readability and maintainability.